kickstarter/android-oss
# View my pledge

# What ❓
- Created a color state list for ended projects `button_pledge_ended`.
- Renamed the color state list for live projects `button_pledge_live`.
- Removed the overscroll from the rewards list (the green alpha at the beginning and end).
- No rewards
  - There is a bug (IMO) with the `project.backing.rewardId`. It returns null, but the no reward `reward.id` is `0`.  I updated `BackingUtils.isBacked` to handle that case (see tests).
  - This VM and VH need to be refactored because they don't match the Abstract. Will be done in a separate PR.
- All rewards
  - When the project has ended, the pledge button is not visible unless the user has backed that reward. The days left and remaining amount are not displayed. If it was backed, the pledge button is black and says `View my reward`. The checkmark is visible.
  - Refactored `HorizontalRewardViewHolderViewModel` because several outputs were unused and didn't have tests.
- Added `ProjectUtils.pledgeButtonColor` that returns the color resource ID of the rewards button based on project and backing status.
- Rewards are now free scrolling with `18dp` between them.
- Added the output `backedRewardPosition` to `RewardFragmentViewModel` so the backed reward is centered in the rewards list when the rewards sheet is opened on an ended project.
- Added `RewardUtils.isExpired` and `isTimeLimited` to handle logic for showing when a reward is no longer available.
- Added `RewardUtils.pledgeButtonColor` that returns the color resource ID of the rewards button based on project and if user has backed reward.

# How to QA? 🤔
View an ended project that you've backed (check your profile). (Note: If you click `View my pledge`, the screen never loads. This is a bug that has been present for a while in production but no one has said anything 🤫. It'll be fixed in a separate PR.)

# Story 📖
[Trello Story](https://trello.com/c/gu8urriv/1414-reward-state-view-my-pledge)

# See 👀
![device-2019-06-14-182136 2019-06-14 18_23_12](https://user-images.githubusercontent.com/1289295/59541922-7ee8b600-8ed1-11e9-9fc6-c77b52cd2eee.gif)
